### PR TITLE
Don't mark "safe" intrinsics as unsafe

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
A good example of this is [intrinsics::abort](https://doc.rust-lang.org/nightly/core/intrinsics/fn.abort.html).

Before:

Screenshot from 2021-06-15 14-58-42

After:

Screenshot from 2021-06-15 14-59-22

cc @jyn514 
r? @lqd

### CODE REVIEW

**Comment by jyn514** [comment]
Can you add a test for this?

**Comment by GuillaumeGomez** [comment]
Well, I have no idea how to do that... Only a few specific intrinsics listed in the typeck function are safe to call, and we don't run DOM tests on std/core. If you have an idea by any chance?

**Comment by jyn514** [comment]
I thought intrinsics were part of the compiler, not the standard library?

Anyway, you can replicate however the standard library does it in a standalone file with #![no_core] and then test that.

**Comment by GuillaumeGomez** [comment]
Oh good idea!

**Comment by GuillaumeGomez** [comment]
Done, thanks again for the tip!

**Comment by lqd** [comment]
Nicely done. The test is exactly what I was hoping for (thanks @jyn514).

Thanks for taking care of it so quickly !
@bors r+

File: src/librustdoc/clean/mod.rs [hunk 1]
```diff
@@ -2132,7 +2133,7 @@ impl Clean<Item> for (&hir::ForeignItem<'_>, Option<Symbol>) {
                         decl,
                         generics,
                         header: hir::FnHeader {
-                            unsafety: hir::Unsafety::Unsafe,
+                            unsafety: intrinsic_operation_unsafety(item.ident.name),
```
Comment (bjorn3) [inline src/librustdoc/clean/mod.rs:2136]: This should only apply for `extern "rust-intrinsic"`, not `extern "C"`.

Comment (GuillaumeGomez) [reply src/librustdoc/clean/mod.rs:2136]: Good point, thanks for bringing it up!

**Comment by GuillaumeGomez** [comment]
I added the abi check.

**Comment by lqd** [comment]
Could we add something like this to the test ? [`core::intrinsics::needs_drop`](https://github.com/rust-lang/rust/blob/246571a846e29691bd535d531365acd318857207/compiler/rustc_typeck/src/check/intrinsic.rs#L62) is [safe as well](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=3abc8f1cfde603483556a6018f8d233f) 

```rust
extern "C" {
    // @has 'foo/fn.needs_drop.html'
    // @has - '//pre[@class="rust fn"]' 'pub unsafe extern "C" fn needs_drop() -> !'
    pub fn needs_drop() -> !;
}
```

**Comment by GuillaumeGomez** [comment]
@lqd Added your test as well!

**Pull request merged**